central/complianceoperator/v2/scanconfigurations/service/service_impl.go

```diff
@@ -220,6 +227,38 @@ func (s *serviceImpl) RunComplianceScanConfiguration(ctx context.Context, reques
 	return &v2.Empty{}, err
 }
 
+func (s *serviceImpl) RunReport(ctx context.Context, request *v2.ComplianceRunReportRequest) (*v2.ComplianceRunReportResponse, error) {
+	if !features.ComplianceReporting.Enabled() {
+		return nil, errors.Wrap(errox.NotImplemented, "Not implemented")
+	}
+	if request.GetScanConfigId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Scan configuration ID is required to run an a report")
+	}
+
+	scanConfig, found, err := s.scanConfigDS.GetScanConfiguration(ctx, request.GetScanConfigId())
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve compliance scan configuration with id %q.", request.GetScanConfigId())
+	}
+	if !found {
+		return nil, errors.Errorf("failed to retrieve compliance scan configuration with id %q.", request.GetScanConfigId())
+	}
+
+	err = s.reportManager.SubmitReportRequest(ctx, scanConfig)
+	if err != nil {
+		return &v2.ComplianceRunReportResponse{
+			RunState:    v2.ComplianceRunReportResponse_ERROR,
+			SubmittedAt: types.TimestampNow(),
+			ErrorMsg:    err.Error(),
+		}, errors.Wrapf(err, "failed to submit compliance on demand report request for scan config %q", scanConfig.GetScanConfigName())
```

Comment:
nit: and not sure the suggestion is any better.

```suggestion
		}, errors.Wrapf(err, "unable to generate a compliance on demand report for scan config %q", scanConfig.GetScanConfigName())
```

Reply:
Aah, so keeping my version because I believe the report generation errors will have that verbiage - trying to distinguish between the act of submitting a report request and the actual report generation work for that request.